code/modules/mob/living/death.dm

```diff
@@ -96,3 +96,9 @@
 
 	// u no we dead
 	return TRUE
+
+/mob/living/proc/delayed_gib()
+	visible_message("<span class='danger'><b>[src]</b> starts convulsing violently!</span>", "You feel as if your body is tearing itself apart!")
+	Weaken(15)
+	do_jitter_animation(1000, -1)
+	addtimer(CALLBACK(src, .proc/gib), rand(20, 100))
```

Comment:
@KasparoVy 

Since you seem interested in this proc, the beauty of `addtimer` is that if the mob gets deleted before this can happen, it won't generate a runtime.

Reply:
Super safe! Man that's brilliant. Thanks for tagging me here, I appreciate the explanation and example.

I'm definitely seeing how big a step forward these functions are for SS13 code